sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/JsonNodeStorePayload.java

```diff
@@ -40,4 +71,39 @@ public int getResponsePayloadSize() {
     public JsonNode getPayload() {
         return jsonValue;
     }
+
+    private static CharsetDecoder getFallbackCharsetDecoder() {
+        if (StringUtil.isNullOrEmpty(Configs.getCharsetDecoderErrorActionOnMalformedInput())
+         && StringUtil.isNullOrEmpty(Configs.getCharsetDecoderErrorActionOnMalformedInput())) {
+            logger.debug("No fallback charset decoder is enabled");
+            return null;
+        }
+
+        CharsetDecoder charsetDecoder = StandardCharsets.UTF_8.newDecoder();
+        // config coding error action for malformed input
+        switch (Configs.getCharsetDecoderErrorActionOnMalformedInput().toUpperCase()) {
+            case "REPLACE":
+                charsetDecoder.onMalformedInput(CodingErrorAction.REPLACE);
+                break;
+            case "IGNORE":
+                charsetDecoder.onMalformedInput(CodingErrorAction.IGNORE);
+                break;
+            default:
+                break;
```

Comment:
For this, I recommend adding a warning log, letting customer know that they have chosen an incorrect error action for malformed charset. Otherwise, they might never know if they have a typo or don't know if the error action was applied or not.

Reply:
added warning log